src/pages/PopupPage.js

```diff
@@ -178,22 +219,152 @@ function ApproveConnectionForm({ origin, onApprove }) {
           <Typography>{wallet.publicKey.toBase58()}</Typography>
         </div>
         <Typography>Only connect with sites you trust.</Typography>
+        <Divider className={classes.divider} />
+        <FormControlLabel
+          control={
+            <Switch
+              checked={autoApprove}
+              onChange={() => setAutoApprove(!autoApprove)}
+              color="primary"
+            />
+          }
+          label={`Automatically approve Serum DEX transactions`}
+        />
+        {!dismissed && autoApprove && (
+          <SnackbarContent
+            className={classes.warningContainer}
+            message={
+              <div>
+                <span className={classes.warningTitle}>
+                  <WarningIcon className={classes.warningIcon} />
+                  Use at your own risk.
+                </span>
+                <Typography className={classes.warningMessage}>
+                  This setting allows {origin} to send standard Serum DEX
+                  transactions on your behalf without requesting your permission
+                  for the remainder of this session.
+                </Typography>
+              </div>
+            }
+            action={[
+              <Button onClick={() => setDismissed('1')}>I understand</Button>,
+            ]}
+            classes={{ root: classes.snackbarRoot }}
+          />
+        )}
       </CardContent>
       <CardActions className={classes.actions}>
         <Button onClick={window.close}>Cancel</Button>
-        <Button color="primary" onClick={onApprove}>
+        <Button
+          color="primary"
+          onClick={() => onApprove(autoApprove)}
+          disabled={!dismissed && autoApprove}
+        >
           Connect
         </Button>
       </CardActions>
     </Card>
   );
 }
 
-function ApproveSignatureForm({ origin, message, onApprove, onReject }) {
+function isSafeInstruction(publicKeys, owner, instructions) {
+  let unsafe = false;
+  const states = {
+    CREATED_AND_OWNED: 0,
+    INITIALIZED_AND_OWNED: 1,
+    CLOSED_TO_OWNED_DESTINATION: 2,
+  };
+  const newAccounts = {};
+  const initializedAccountMints = {};
+
+  function isOwned(pubkey) {
+    if (!pubkey) {
+      return false;
+    }
+    if (
+      publicKeys &&
+      publicKeys.some((ownedAccountPubkey) => ownedAccountPubkey.equals(pubkey))
```

Comment:
this can be `publicKeys?.some(...)` instead of `publicKeys && publicKeys.some(...)`